tests/func/utils/test_strict_yaml.py

```diff
@@ -351,7 +351,11 @@ def print_with_fixed_width(*args, **kwargs):
     out, err = capsys.readouterr()
 
     assert not out
-    assert expected in err
+
+    # strip whitespace on the right: output is always left-justified
+    # by rich.syntax.Syntax:
+    for l_expected, l_err in zip(expected.split("\n"), err.split("\n")):
+        assert l_expected.rstrip(" ") == l_err.rstrip(" ")
```

Comment:
Let's use `splitlines`, and not use `rstrip()` in the expected value.
```suggestion
    for l_expected, l_err in zip(expected.splitlines(), err.splitlines()):
        assert l_expected == l_err.rstrip(" ")
```

Reply:
Few nitpicky suggestions:

1. I prefer `_line` suffix than `l_` prefix.
2. Keep `expected` value on the right hand side.
```suggestion
    for err_line, expected_line in zip(err.splitlines(), expected.splitlines()):
        assert err_line.rstrip() == expected_line
```

But no strong opinion, just being nitpicky here.

Reply:
Note that using `rstrip()` instead of `rstrip(" ")` will also strip newlines

Reply:
> Note that using rstrip() instead of rstrip(" ") will also strip newlines

@dtrifiro, sorry, you are right. I was confusing it with `bytes.rstrip()`.